### ISSUE TITLE
Create Doc Home Page

### ISSUE THREAD

**Issue by jbogarthyde** [in-toto/docs#86]
To be immediately useful, the landing page should provide a top-level roadmap to [existing docs](https://github.com/cncf/techdocs/blob/main/assessments/0009-in-toto/survey-of-existing-doc.md). The roadmap must be updated each time documents are added or moved.

The landing page URL for the [read-the-docs site](https://in-toto.readthedocs.io/en/latest/) currently hosts the auto-generated Python reference doc. To use Read the Docs as the site generator, expand and repurpose this URL as the new overall Doc home page, add user-doc build files in user-doc source directory, and modify the destination for the reference-doc build. 

   - [ ] Add a *source* subdirectory to https://github.com/in-toto/docs to contain the sources for web-based user documentation. 
   - [ ] Add a new source file for the Documentation Home Page. The initial content is roadmap that describes and provides access to:
      - in-toto Technical Specification 
      - Basic demo
      - Python reference implementation reference docs
    - [ ] Add  build files in the [docs repo](https://github.com/in-toto/docs) for the chosen static site generation tool to build the new doc home page.
    - [ ] Change the destination for the reference doc build. It should go into a sub-section of the home page with the title "Python Reference Documentation"). The makefile and sources for the generated doc are in https://github.com/in-toto/in-toto/tree/develop/doc
    - [ ]  Add a link to the new Doc Home Page. Make it the first menu item in the Learn More menu on the [project home page](https://in-toto.io/). 

See [Umbrella Doc Issue](https://github.com/in-toto/docs/issues/85#top)

**Comment by jbogarthyde** [comment]
Please add the documentation label to this issue

**Comment by DarikshaAnsari** [comment]
Hey @JustinCappos, 
As mentioned in the issue description we just have to utilize the current source code for the ` read-the-doc` and modify it so that a dedicated documentation homepage can be served to the users?

If you think I am going in the right direction, It would be motivating if you can assign this issue to me.

**Comment by DarikshaAnsari** [comment]
@lukpueh No, I think we don't have to change complete content of this repo, rather we have to only add source subdirectory and make it visible on the first menu item in the Learn More menu on `in-toto.io`.
Also for the existing users the current read-the-doc contain Python reference implementation, which should go into a sub-section of the new home page with the title "Python Reference Documentation" , which will not confuse existing user.

**Comment by lukpueh** [comment]
> @lukpueh No, I think we don't have to change complete content of this repo, rather we have to only add source subdirectory and make it visible on the first menu item in the Learn More menu on `in-toto.io`.

Sure, we don't completely change the content, but we further broaden the scope of the repo. This makes sense if you think of the repo as what it's currently called, i.e. a general "docs" repo. But it makes less sense, if you think of it as what it actually is, i.e. the "spec" repo. IMO the specification is important enough to have its own dedicated repo, with an intuitive name, e.g. "specification".

So one idea is to just carve out the spec documents, dump them to a new repo, and keep this one the general docs repo.  
I think that's fine too, the only downside is that there likely are many links to the spec in this repo out there, which would break. Now, if we rename the repo, people looking for the spec in this repo, will be forwarded to the new repo.

This would leave us without a "docs" repo for general docs. But I think in-toto.io could fill that spot.

> Also for the existing users the current read-the-doc contain Python reference implementation, which should go into a sub-section of the new home page with the title "Python Reference Documentation" , which will not confuse existing user.

Same as above, I think moving things around like this will be more disruptive than beneficial.

**Comment by lukpueh** [comment]
Another reason for using in-toto.io for general docs is to bundle our efforts in one place. in-toto.io could already use more maintainers/contributors. If we have two places for general project-wide infos and docs, this situation will probably not get better.

**Comment by DarikshaAnsari** [comment]
but we already have one file `in-toto-spec.md` which already contain all the necessary specification of the in-toto which is currently reside in in-toto doc [here](https://github.com/in-toto/docs/blob/master/in-toto-spec.md), so if we make this repo as specification repo so there is a confusion between these two as we have two file/repo which describe the specification of in-toto

**Comment by lukpueh** [comment]
I think there's a misunderstanding. My proposal above is to [rename](https://docs.github.com/en/repositories/creating-and-managing-repositories/renaming-a-repository#) `github.com/in-toto/docs` to `github.com/in-toto/specification`.

This means there no longer will be a  `github.com/in-toto/docs` repo, but old links to files in that repo (e.g. in-toto-spec.md) still work, as they will automatically be redirected to the same file in  `github.com/in-toto/specification`.

**Comment by DarikshaAnsari** [comment]
Yeah, I got your point now , we can use in-toto.io for general docs makes it easier for new users and rename this repo as specs repo for developers/contributor so that new users gets helpful information without sorting through technical specifications.

**Comment by DarikshaAnsari** [comment]
Yes!! I am interested working on this.
Let me raise an issue [here](https://github.com/in-toto/in-toto.io/issues) highlighting the following key aspects by referencing about the current issue [#86](https://github.com/in-toto/docs/issues/86):

1. **Objective of raising the issue**
2. ** Proposed Changes**
3. ** Expected Outcome**

We can have a dedicated discussion regarding the further enhancement or your reviews over the proposed changes in the issue I will raise against [ https://github.com/in-toto/in-toto.io/issues ]( https://github.com/in-toto/in-toto.io/issues )

**Comment by jbogarthyde** [comment]
I believe the easiest way to proceed is to create the docs home page  as the landing page **in-toto.io**. This URI currently  lands on the reference doc for Python Reference Implementation. That doc is logically part of "project-wide docs", so it should be included in the RTD project, but moved down to a subdir, instead of being the landing page. 

On the home page, there should be a TOC for the "project-wide docs". The TOC should have a section for Implementation Reference Docs, with a link to the Python Reference Implementation reference docs (at their new location), along with links to the reference docs for other implementations (which might be external).

I hope this is clear.

**Comment by lukpueh** [comment]
> I believe the easiest way to proceed is to create the docs home page as the landing page **in-toto.io**. This URI currently lands on the reference doc for Python Reference Implementation.

Did you mean **http://in-toto.readthedocs.io**? **in-toto.io** currently does not land on the reference doc for Python in-toto.

> That doc is logically part of "project-wide docs", so it should be included in the RTD project, but moved down to a subdir, instead of being the landing page.

There has been an agreement in the in-toto community that the Python "reference" implementation should be considered as one of multiple in-toto implementations. That is why I suggested to keep the project-wide docs (in-toto.io) and Python in-toto docs (http://in-toto.readthedocs.io, we can change the name to python-in-toto.readthedocs.io) separate. 

> 
> On the home page, there should be a TOC for the "project-wide docs". The TOC should have a section for Implementation Reference Docs, with a link to the Python Reference Implementation reference docs (at their new location), along with links to the reference docs for other implementations (which might be external).

Agreed! (except for the part about new location for Python in-toto)

> 
> I hope this is clear.

Not entirely. I'd appreciate a comment to my question in https://github.com/in-toto/docs/issues/85#issuecomment-2107148234, that is, whether there is a specific reason to not use **in-toto.io** for project-wide docs.

**Comment by jbogarthyde** [comment]
Thanks for the correction, Lukas -- yes, I did indeed mean that the doc home page should be http://in-toto.readthedocs.io/, with a prominent "Documentation" link on the in-toto home page, http://in-toto.io, and another link at the top of the Get Started menu. 

The project-wide doc home page can point to a Reference Docs section, with a link to the Python Reference Implementation reference docs. The sources for that doc can remain in Read-the-docs, but would need a new landing page - something like  http://in-toto.readthedocs.io/referencedocs/python. 

The point of using http://in-toto.readthedocs.io/ as the doc landing page (rather than, say, in-toto.io/docs) is to keep RTD as the doc development tool, thus avoiding the overhead of starting with a new tool.

**Comment by lukpueh** [comment]
> The point of using http://in-toto.readthedocs.io/ as the doc landing page (rather than, say, in-toto.io/docs) is to keep RTD as the doc development tool, thus avoiding the overhead of starting with a new tool.

Okay, thanks for clarifying! Given that we actually don't need to start a new tool, but can use the existing in-toto.io static site generator, plus the [other arguments](https://github.com/in-toto/docs/issues/86#issuecomment-2102244076) for using in-toto.io and against using in-toto.readthedocs.io for project-wide docs, I suggest we implement Judy's plan (#85) in in-toto.io. @DarikshaAnsari has kindly created an issue there: https://github.com/in-toto/in-toto.io/issues/24#issuecomment-2107194347
